plugwise/smile.py

```diff
@@ -131,15 +133,21 @@ async def async_update(self) -> PlugwiseData:
         # Perform a full update at day-change
         self.gw_data: GatewayData = {}
         self.gw_devices: dict[str, DeviceData] = {}
-        await self.full_update_device()
-        self.get_all_devices()
 
-        if "heater_id" in self.gw_data:
-            self._heater_id = self.gw_data["heater_id"]
-            if "cooling_enabled" in self.gw_devices[self._heater_id]["binary_sensors"]:
-                self._cooling_enabled = self.gw_devices[self._heater_id]["binary_sensors"]["cooling_enabled"]
+        try:
+            await self.full_update_device()
+            self.get_all_devices()
+            if "heater_id" in self.gw_data:
+                self._heater_id = self.gw_data["heater_id"]
+                if "cooling_enabled" in self.gw_devices[self._heater_id]["binary_sensors"]:
+                    self._cooling_enabled = self.gw_devices[self._heater_id]["binary_sensors"]["cooling_enabled"]
 
-        return PlugwiseData(self.gw_data, self.gw_devices)
+            return PlugwiseData(self.gw_data, self.gw_devices)
+        except:
```

Comment:
Should we except `KeyError` as you spotted? As indicated by codefactor a bare except shouldn't be there

Reply:
Solved.